lib/generator.js

```diff
@@ -72,15 +72,28 @@ const generateFile = options => new Promise((resolve, reject) => {
         const template_path = path.relative(templates_dir, path.resolve(root, file_name));
         const generated_path = path.resolve(target_dir, template_path);
 
-        fs.writeFile(generated_path, parsed_content, 'utf8', (err) => {
-          if (err) return reject(err);
-          resolve();
+        stat(source_path)
+          .then(stats => {
+            fs.writeFile(generated_path, parsed_content, {encoding: 'utf8', mode: stats.mode}, (err) => {
```

Comment:
@fmvilas there is probably a cleaner way to do this. I mean the promise inside another one